#!/usr/bin/env python
PACKAGE = "roboclaw"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("pid_p", int_t, 0, "PID Proportional gain", 15000, 0, 0x10000)
gen.add("pid_i", int_t, 0, "PID Integral gain", 0x250, 0, 0x1000)
gen.add("pid_d", int_t, 0, "PID Derivative gain", 500, 0, 0x1000)

gen.add("pid_qpps", int_t, 0, "Quad pulses per second", 300000, 0, 500000)
gen.add("accel_max", double_t, 0, "Maximum wheel acceleration (m/s^2)",
        1.0, 0, 2.0)
gen.add("max_wheel_vel", double_t, 0, "Maximum individual wheel speed (m/s)",
        0.8, 0, 1.0)
gen.add("min_wheel_vel", double_t, 0, "Minimum individual wheel speed (m/s)",
        0.0, 0, 1.0)
gen.add("wheel_diam", double_t, 0, "Wheel diameter (m)",
        0.1, 0.01, 0.5)
gen.add("quad_pulse_per_motor_rev", int_t, 0, "Quad pulses / motor revolution",
        2000, 0, 5000)
gen.add("motor_to_wheel_ratio", double_t, 0,
        "Number of motor revolutions for one wheel revolution",
        40, 0.01, 100)
gen.add("axle_width", double_t, 0, "Spacing between wheels (m)",
        0.255, 0.01, 0.4)

gen.add("freq", double_t, 0, "The update rate of odometry integration (hz)",
        30, 0, 100)

gen.add("odom_frame", str_t, 0, "The odometry frame of the robot", "odom")
gen.add("base_frame", str_t, 0, "The local frame of the robot", "base")


exit(gen.generate(PACKAGE, "roboclaw_node", "Roboclaw"))
